lib/Data/Chronicle/Writer.pm

```diff
@@ -128,38 +140,80 @@ publish "category1::name1" in Redis if C<publish_on_set> is true.
 =cut
 
 sub set {
-    my $self     = shift;
-    my $category = shift;
-    my $name     = shift;
-    my $value    = shift;
-    my $rec_date = shift;
-    my $archive  = shift // 1;
-    my $ttl      = shift // $self->ttl;
+    my $self         = shift;
+    my $category     = shift;
+    my $name         = shift;
+    my $value        = shift;
+    my $rec_date     = shift;
+    my $archive      = shift // 1;
+    my $suppress_pub = shift // 0;
+    my $ttl          = shift // $self->ttl;
 
-    die "Recorded date is undefined" unless $rec_date;
-    die "Recorded date is not a Date::Utility object" if ref $rec_date ne 'Date::Utility';
-    die "Cannot store undefined values in Chronicle!" unless defined $value;
-    die "You can only store hash-ref or array-ref in Chronicle!" unless (ref $value eq 'ARRAY' or ref $value eq 'HASH');
+    $self->mset([[$category, $name, $value]], $rec_date, $archive, $suppress_pub, $ttl);
+
+    return 1;
+}
+
+=head2 mset
+
+Example:
+
+    $chronicle_writer->mset([["category1", "name1", $value1], ["category2, "name2", $value2], ...]);
 
-    $value = JSON::MaybeXS->new->encode($value);
+Store a piece of data "value1" under key "category1::name1", "category2::name2", etc in Pg and Redis. Will
+publish "category1::name1", "category2::name2", etc in Redis if C<publish_on_set> is true.
+
+=cut
+
+sub mset {
+    my $self         = shift;
+    my $entries      = shift;
+    my $rec_date     = shift;
+    my $archive      = shift // 1;
+    my $suppress_pub = shift // 0;
+    my $ttl          = shift // $self->ttl;
+
+    $self->_validate_value($_->[2]) foreach @$entries;
+    $self->_validate_rec_date($rec_date);
 
-    my $key    = $category . '::' . $name;
     my $writer = $self->cache_writer;
 
     # publish & set in transaction
     $writer->multi;
-    my $encoded = encode_utf8($value);
-    $writer->publish($key, $encoded) if $self->publish_on_set;
-    $writer->set(
-        $key => $encoded,
-        $ttl ? ('EX' => $ttl) : ());
-    $writer->exec;
+    foreach my $entry (@$entries) {
+        my $category = $entry->[0];
+        my $name     = $entry->[1];
+        my $value    = $entry->[2];
+
+        my $key = $category . '::' . $name;
 
-    $self->_archive($category, $name, $value, $rec_date) if $archive and $self->dbic;
+        my $encoded = encode_json_utf8($value);
+        $writer->publish($key, $encoded) if $self->publish_on_set && !$suppress_pub;
+        $writer->set(
+            $key => $encoded,
+            $ttl ? ('EX' => $ttl) : ());
+
+        $self->_archive($category, $name, $encoded, $rec_date) if $archive and $self->dbic;
+    }
+    $writer->exec;
 
     return 1;
 }
 
+sub _validate_value {
+    my ($self, $value) = @_;
```

Comment:
With the above suggestion, this can now become something like

    my ($self, $category, $name, $value) = @_;
    die "Cannot store an undefined value for ${category}::${name} in Chronicle!" ...